src/less/button/button.less

```diff
@@ -354,7 +354,10 @@ a.fake-btn--form {
     border-color: inherit;
     .border-radius-token(expand-btn-border-radius, border-radius-50);
     max-width: 100%;
+}
 
+button.btn--form:not([disabled], [aria-disabled="true"]),
```

Comment:
If we are including buttons in this change as well, why is this only applied to `btn--form` modified buttons? I'm not sure button outside of forms should be treated differently.

Reply:
Agree that buttons should not be treated any difference. But the way the code is structured today, disabled styles are applied at modifier level. For e.g.,

```
button.btn--primary.btn--destructive[disabled],
button.btn--primary.btn--destructive[aria-disabled="true"] {}

button.btn--primary.btn--destructive:not([disabled], [aria-disabled="true"]),
a.fake-btn--primary.fake-btn--destructive[href] {}
```

When we added `btn--form` with overrides, its overriding above styles with higher specificity. So, I added disabled styles for form modifiers as well to keep it consistent.